### PR TITLE
Script for creating a coverage report

### PR DESCRIPTION
**Pull request by emaxx-google**
Add a script "coverage-report.sh" that builds the code in the
TOOLCHAIN=coverage mode, merges the collected raw profiles (*.profraw),
and creates the resulting coverage report.

### CODE REVIEW

File: coverage-report.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+UNIT_TEST_EXECUTABLE_DIRS="
+  common/cpp/build/tests
+  third_party/libusb/webport/build/tests"
+IGNORE_FILENAME_REGEX="
+  third_party/googletest"
+
+log_message() {
+  local message=${1}
+
+  echo -e "\033[33;32m******* ${message} *******\033[0m"
```
Comment (Fabian-Sommer) [inline coverage-report.sh:34]: What does this line do (apart from printing the message)? I would guess it is some coloring for the shell, but it is very cryptic.

**Review by Fabian-Sommer — APPROVED** [review]

File: coverage-report.sh (hunk 1 above)
Comment (emaxx-google) [reply coverage-report.sh:34]: Done - Added a comment. The syntax is cryptic indeed, but I don't know an easier way (and coloring helps read the logs significantly).

**Review by Fabian-Sommer — APPROVED** [review]

File: coverage-report.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+COVERAGE_PROFILE_DIRS="
+  common/cpp/build/tests
+  third_party/libusb/webport/build/tests"
+EXECUTABLE_MODULE_DIRS="
+  smart_card_connector_app/build/executable_module"
+UNIT_TEST_EXECUTABLE_DIRS="
+  common/cpp/build/tests
+  third_party/libusb/webport/build/tests"
+IGNORE_FILENAME_REGEX="
+  third_party/googletest"
```
Comment (emaxx-google) [inline coverage-report.sh:29]: BTW, should we also exclude the unit test files? ("*test.cc")
Not sure what's the common practice for calculating the overall coverage percentage.

Comment (kaszarek) [reply coverage-report.sh:29]: Unit tests provide coverage as well. 
IMHO we could start with having them and we will see if some lines are too exhaustively covered. Then, if needed, we will take it from there.

**Review by kaszarek — APPROVED** [review]
LGTM

**Pull request merged**